### ISSUE TITLE
Feature Request: Return promises in asynchronous API functions

### ISSUE THREAD

**Issue by jugador-argentino** [haxball/haxball-issues#1952]
There are many times when I want to know when a function call triggers an event. 

For example, setCustomStadium, you can call the function multiple times, but the event trigger is canceled if the game is started, so I can't keep track between the function call and the triggered event.

Another example is when I want to wait for all the calls to finish executing, there could be a call that hasn't been executed yet and I couldn't know it.

**Comment by mertushka** [comment]
Good point, especially `setPlayerTeam` and `getPlayerList` functions gets badly affected by asynchronous.
